### DISCUSSION TITLE
Changing hand finger thickness

### DISCUSSION THREAD

**Discussion by paulmelis** [StereoKit/StereoKit#337] [Q&A] [answered]
I've just been trying to use StereoKit with a Varjo XR-3 headset and I'm amazed that it works out of the box, even in mixed reality mode! A lot of that is thanks to OpenXR and its support in SK, great job!

One immediate issue is that the virtual fingers shown (based on UltraLeap hand tracking) are really fat and I'd like to scale them down a bit. I don't quite see a property for that in the SK API, should there be? Or does finger thickness come from the tracking info?

**Comment by maluoi** [comment]
There is actually a decent Varjo simulator that I've run SK through, so thanks in part to their dev tools as well! I added `SKSettings.blendPreference` in particular based on that experience to work with Varjo passthrough type devices :)

OpenXR does directly provide its own finger joint radius, and StereoKit just uses what it's given. I don't think I've actually seen Leap hand tracking on a "transparent" display, so I've never been able to judge the joint size relative to the real hand size! I've definitely had the impression that Leap fingers are a bit larger than those provided by others though. I'm not certain where's the _best_ place to bring this up, but it might be worth mentioning over on their [OpenXR Layer GitHub page](https://github.com/ultraleap/OpenXRHandTracking)? I'd definitely be curious what they say about that.

**Reply by paulmelis** [reply]
> I'm not certain where's the best place to bring this up, but it might be worth mentioning over on their [OpenXR Layer GitHub page](https://github.com/ultraleap/OpenXRHandTracking)? I'd definitely be curious what they say about that.

It seems you already made a remark on this a year ago: https://github.com/ultraleap/OpenXRHandTracking/issues/2 😃 

I'll ping that issue to see if there are any updates.

**Reply by maluoi** [reply]
Oh goodness, I forgot all about that! Fancy that :D

**Comment by paulmelis** [comment]
I also asked Varjo directly about this issue, will report back here when I get an answer.

**Comment by paulmelis** [comment]
So, this is still an ongoing issue, and I'm not really getting anywhere in contacts with Varjo and Ultraleap (and probably a low priority for them to get this fixed). Do you think your recent updates in using XR_EXT_hand_tracking_data_source will make an influence here, as I noticed some remarks in the code on Ultraleap specifically?

If not, I would like to make a patch to add a finger thickness scaling factor, so I at least can work around the issue. But I have no idea where to start in the code, so any pointers you have would be great.

Edit: the main issue is #425, button misclicks due to finger thickness

**Reply by maluoi** [reply, marked as answer]
`XR_EXT_hand_tracking_data_source` unfortunately will have no impact on this! I did just add a `Backend.OpenXR.SetHandJointScale` though over in #669, so you should have that in the next build tonight. I'll see about asking Ultraleap about the joints as well :)

**Reply by paulmelis** [reply]
Oh, that's neat, thanks!
